tools/fog-local-network/fog_conformance_tests.py

```diff
@@ -564,23 +557,16 @@ def run(self):
         wallet3_from5 = self.fresh_balance_check("from5", 3, {4: 20, 5: 0}, [5])
         wallet4_from5 = self.fresh_balance_check("from5", 4, {4: 1, 5: 0}, [5])
 
-        self.follow_up_balance_check(wallet0_from1, {5: 10}, [5])
-        self.follow_up_balance_check(wallet1_from1, {5: 6}, [5])
-        self.follow_up_balance_check(wallet2_from1, {5: 0}, [5])
-        self.follow_up_balance_check(wallet3_from1, {5: 0}, [5])
-        self.follow_up_balance_check(wallet4_from1, {5: 0}, [5])
-
-        self.follow_up_balance_check(wallet0_from4, {5: 10}, [5])
-        self.follow_up_balance_check(wallet1_from4, {5: 6}, [5])
-        self.follow_up_balance_check(wallet2_from4, {5: 0}, [5])
-        self.follow_up_balance_check(wallet3_from4, {5: 0}, [5])
-        self.follow_up_balance_check(wallet4_from4, {5: 0}, [5])
-
-        self.follow_up_balance_check(wallet0_from4a, {5: 10}, [5])
-        self.follow_up_balance_check(wallet1_from4a, {5: 6}, [5])
-        self.follow_up_balance_check(wallet2_from4a, {5: 0}, [5])
-        self.follow_up_balance_check(wallet3_from4a, {5: 0}, [5])
-        self.follow_up_balance_check(wallet4_from4a, {5: 0}, [5])
+        for (wallet0, wallet1, wallet2, wallet3, wallet4) in (
+            (wallet0_from1, wallet1_from1, wallet2_from1, wallet3_from1, wallet4_from1),
+            (wallet0_from4, wallet1_from4, wallet2_from4, wallet3_from4, wallet4_from4),
+            (wallet0_from4a, wallet1_from4a, wallet2_from4a, wallet3_from4a, wallet4_from4a),
+        ):
```

Comment:
interesting -- i did not know this works in python

one thing is, when the test fails, you won't get a line number showing which one it is, but it there is a "name" for the wallet like wallet3_from1 that gets reported when balance check reports an error

Reply:
I'll look into ensuring the name associated with the wallets is printed before each check.